app/models/profile_link.rb

```diff
@@ -26,11 +26,18 @@ class ProfileLink < ApplicationRecord
   belongs_to :profile_link_site, required: true
 
   validates :url, presence: true
+  validate :updated_url, if: :url
 
   before_validation do
-    self.url = url&.gsub(
-      Regexp.new(profile_link_site.validate_find),
+    # updates the url to the expected output
+    # Twitter, ie: toyhammered -> https://twitter.com/toyhammered
+    self.url = url&.sub(
+      profile_link_site.validate_find,
       profile_link_site.validate_replace
     )
   end
+
+  def updated_url
```

Comment:
I wasn't actually sure what to name it (because I suck at naming) but it is checking whether the `updated_url` is valid. Maybe it should be something like `reformatted_url`?